src/jdk.internal.vm.ci/share/classes/jdk/vm/ci/meta/ResolvedJavaField.java

```diff
@@ -40,6 +40,13 @@ public interface ResolvedJavaField extends JavaField, ModifiersProvider, Annotat
     @Override
     int getModifiers();
 
+    /**
+     * {@inheritDoc}
+     * <p>
+     * Returns VM internal flags associated with this field
+     */
+    int getInternalModifiers();
```

Comment:
We've never exposed the internal modifiers before in public JVMCI API and we should refrain from doing so until there's a good reason to do so. Please remove this method.

Reply:
Access to internal modifiers is needed in `HotSpotResolvedJavaFieldTest.testEquivalenceForInternalFields()`. I moved the declaration of the method to `HotSpotResolvedJavaField`. Does this change work for you?

Reply:
Just use reflection to read the internal flags (like this test already does for the `index` field).

I've attached [review.patch](https://github.com/openjdk/jdk/files/10970245/review.patch) with this change and a few other changes I think should be made for better naming (plus one test cleanup).

Reply:
Thank you for the patch, it will be included in the next commit.